### PR TITLE
docs(code-examples): add data-grid paging example

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2442598](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2442598)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1091/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/1091/storybooks/a11y-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1091/storybooks/ag-grid-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/1091/storybooks/core-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/1091/storybooks/indicators-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1091/storybooks/layout-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/1091/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1091/storybooks/pages-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/1091/storybooks/popovers-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1091/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/1091/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1091/storybooks/tabs-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/1091/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/1091/storybooks/tiles-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1091/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1091/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1091/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1091/playground/)

File: apps/code-examples/src/app/code-examples/ag-grid/data-grid/paging/data-grid-paging-demo.component.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  public gridOptions: GridOptions;
+  public searchText = '';
+
+  #subscriptions = new Subscription();
+
+  constructor(
+    private agGridService: SkyAgGridService,
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private changeDetector: ChangeDetectorRef
```
Comment (Blackbaud-SteveBrush) [inline apps/code-examples/src/app/code-examples/ag-grid/data-grid/paging/data-grid-paging-demo.component.ts]: We could use `#agGridSvc = inject(SkyAgGridService)` etc. for these.

Comment (johnhwhite) [reply apps/code-examples/src/app/code-examples/ag-grid/data-grid/paging/data-grid-paging-demo.component.ts]: Good call. Switched.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick [failed](https://github.com/blackbaud/skyux/actions/runs/4375087890)! :x: Please resolve conflicts and create a new PR.

**Pull request merged**